From 6f7782ceadb10df792cdc4378455d6e03124c770 Mon Sep 17 00:00:00 2001
Message-Id: <6f7782ceadb10df792cdc4378455d6e03124c770.1430046936.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Wed, 1 Apr 2015 17:17:23 +0530
Subject: [PATCH 1/4] SQUASHME! [VPG]: drm/i915: Use generic reg variable for
 BLM_HIST_CTL

Each platform as different BLM_HIST_CTL register the debugfs result might be wrong.
Use the reg variable initilized during dpst reg init.

SQUASHME! - This patch should be squashed into the following existing patch:
commit 548c4386bd0ff14810b0dac7f44faff4120bf28e
change-id: I0eb236fe4ade4065c152ee42952f10e9a207cc77
Author: Daisy Sun <daisy.sun@intel.com>
Date:   Mon May 5 18:58:49 2014 -0700

    MUST_REBASE [VPG]: drm/i915: DPST debugfs read from reg varible

Issue: GMINL-7628
Change-Id: Ia6d55aee4c254d89ab23ac03c8b37ee80133e218
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index 29495df..9038460 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -1862,7 +1862,7 @@ static int i915_dpst_status(struct seq_file *m, void *unused)
 
 	blm_hist_ctl |= BIN_REG_FUNCTION_SELECT_IE;
 	blm_hist_ctl &= ~BIN_REGISTER_INDEX_MASK;
-	I915_WRITE(BLM_HIST_CTL, blm_hist_ctl);
+	I915_WRITE(dev_priv->dpst.reg.blm_hist_ctl, blm_hist_ctl);
 
 	seq_puts(m, "IE modification table values...");
 	for (i = 0; i < DPST_DIET_ENTRY_COUNT; i++) {
-- 
1.7.9.5

